### ISSUE TITLE
Feature Request - Show unused imports TypeScript

### ISSUE THREAD

**Issue by superheri** [vuejs/vetur#1063]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform:  Win 10 1809
- Vetur version: 0.14.5
- VS Code version: 1.31.0-insider

## Problem

Unused imports are not shown in Vue Files like it is in VS Code TypeScrfipt files. Would be nice to have this to know which imports are not used and can be removed.

In a TS file :
![image](https://user-images.githubusercontent.com/30275781/51393455-bf502d80-1b05-11e9-9eb3-bdf703eab59b.png)

In a Vue file : 
![image](https://user-images.githubusercontent.com/30275781/51393516-ddb62900-1b05-11e9-99cc-99aba72d67f6.png)

Even if I never used lodash in my code the import is still shown the same color as other imports unlike in TS files.

## Reproducible Case

Add whatever imports in a Vue files and don't use them.

**Comment by TimeBandit** [comment]
This would help our team too.

**Comment by essenmitsosse** [comment]
Have you tried adding `"noUnusedLocals": true` to `compilerOptions` in your tsconfig? This way the typescript complier handles unused variables (instead of just VS Code). For me this gives me a warning about unused variables and imports, even in .vue files.

**Comment by octref** [comment]
Depends on https://github.com/vuejs/vetur/issues/682. We are using TS 2.8 still and it doesn't have this feature yet.

**Comment by octref** [comment]
Blocked by https://github.com/Microsoft/language-server-protocol/issues/500.

**Comment by octref** [comment]
![image](https://user-images.githubusercontent.com/4033249/55036582-5343e700-4fd8-11e9-825b-e2e0acadca65.png)

Will be available in next version.

For `<script>`, you need to set `checkJs` to true is jsconfig/tsconfig, though.

**Comment by superheri** [comment]
@octref Do we need `checkJs` even if we use tsconfig with `<script lang="ts">`?

**Comment by octref** [comment]
No, `lang="ts"` works OOTB because the script type is `typescript`, so no check "js".

**Comment by superheri** [comment]
Ok. Thanks a lot for this new feature!

**Comment by mengdi** [comment]
@octref It doesn't seem to work in v0.18.0 with <script lang="ts">.
@superheri Did you try this ?

**Comment by octref** [comment]
![image](https://user-images.githubusercontent.com/4033249/55496326-a7ac1f80-55f3-11e9-8fe1-5ada382257b2.png)

**Comment by superheri** [comment]
Does not work on my side too. I don't understand why...

**Comment by octref** [comment]
@mengdi @superheri Can you use either https://github.com/octref/veturpack or vue-cli to create a minimal reproducible example?

**Comment by superheri** [comment]
In my big project at work with ~1250 files of JS/TS/Vue it can't seem to work. But in veturpack it seems to work though. I found a way to produce a bug with veturpack by changing the file `client/views/Home.vue` with this :

```vue
<template>
  <div class="page">
    <counter></counter>
  </div>
</template>

<script lang="ts">
import Counter from 'components/Counter'
import * as _ from 'lodash';
export default {
  components: {
    Counter
  }
}
</script>
```

As we can see in this image (You can use it at the same time as a proof I use Vetur 0.18.0) :

![image](https://user-images.githubusercontent.com/30275781/55521558-37cb8280-564f-11e9-814a-aa275c91eb0d.png)


In this case even if we are not using the declared `_` we still don't see the `not used` message.

I will try hard to reproduce more cases. I'll keep you updated with this.

**Comment by octref** [comment]
Hmm, I just realized that this works:

`"vetur.useWorkspaceDependencies": true` with Veturpack loading TS 2.9.2.

When using bundled TS 3.3 this breaks.

**Comment by octref** [comment]
I see the problem. `lang="ts"` is not recognized as TS file...If you add `checkJS: true` to tsconfig/jsconfig this works.

**Comment by octref** [comment]
We are looking at two issues here:
- https://github.com/vuejs/vetur/issues/1186, which I just fixed.
- https://github.com/Microsoft/TypeScript/issues/30760. If you change the import of lodash from `_` to anything else, Vetur/TSC would start reporting errors. I think the original "feature" was to help people who want to use lodash in a browser (so they could write `_.sort` etc without require/import any lodash code).

**Comment by superheri** [comment]
After testing and trying a way to reproduce the bug I have with my big project, I can't find a way to reproduce it on a minimalist case. Maybe it is a mix of multiple fonctionnalities that creates this bug. Or because it is so slow on big projects that Vetur become unstable.

**Comment by TimeBandit** [comment]
This issues is not resolved for us. We are using `"typescript": "^3.3.4000"` and `<script lang="ts">` the unused imports in our Vue files are now underlined with red.

**Comment by octref** [comment]
Please create a repro case and open new issues.

**Comment by douglasg14b** [comment]
Yeah, this feature doesn't work....

What is the actual resolution here? A reproduction case is literally any project with any .vue file that uses typecript.

**Comment by superheri** [comment]
I can confirm this problems still happens to me. I haven't used Vetur in about a year, but when I got back to it some weeks ago, the problem was there... Seems like a regression at some point. Would be great if it could be fixed...

**Comment by undergroundwires** [comment]
Still does not work in 2021 😢

**Comment by LinusBorg** [comment]
And it likely won't for you in 2022 if people continue to ignore requests by the maintainer for a reproduction.

It seems to work in most cases - works for me, for example. That's also why the issue is closed.

It seems, from what you folks indicate, to break under certain condition - conditions that you, the ones experiencing the problem, have to enable the maintainer to understand and reproduce.

So coming here and posting sad smiley will not help us fix this.

**Comment by undergroundwires** [comment]
Sorry to hear sad smileys, thumbs up from users and recently repeated community feedback don't help, well in that case I hope my dank meme would motivate you until 2022 @LinusBorg 
![image](https://user-images.githubusercontent.com/15555035/104094156-6f894480-528f-11eb-9c42-0e6ed86ce26a.png)

**Comment by LinusBorg** [comment]
the default tsconfig isn't set up to report unused locals. just set it to do so.

**Comment by undergroundwires** [comment]
So the expected behavior is that it does not work without configuring `noUnusedLocals` 🤔. Apparently I have wrongly assumed that it was designed to lint visually regardless of the setting. In my opinion it would be better to not depend on the tsconfig and lint anyway. In the end it's not a bug then, thanks for clearing this out.

**Comment by LinusBorg** [comment]
You can use eslint for that too

**Comment by undergroundwires** [comment]
In my case ESLint without `noUnusedLocals` does not work on ts in `.Vue` files but works in `.ts` files. Repro: `vue create hello-world`

**Comment by yoyo930021** [comment]
Please open a new issue and provide a repro case next time.

I slove this problem in next version, because I just understand this problem.
